### PR TITLE
Fixes the missing borders on customizable homepage cards

### LINKED ISSUE #19368: 🐛 Bug Report: Customizable Homepage Cards missing borders
### 📜 Description

When upgrading from Backstage 1.14 to 1.15 I noticed that the widgets on the customizable homepage grid no longer render correctly (MuiPaper elevation=1). This can also be seen in the [storybook](https://backstage.io/storybook/?path=/story/plugins-home-templates--customizable-template).

### 👍 Expected behavior

The cards should still render the same way they did previously, consistent with the non-customizable homepage.

### 👎 Actual Behavior with Screenshots

It actually looks like this:

1.15+
Screenshot 2023-08-14 at 14 16 44
1.14
Screenshot 2023-08-14 at 14 16 58

### 👟 Reproduction steps

Enable the customizable homepage in Backstage 1.15+.

### 📃 Provide the context for the Bug.

_No response_

### 🖥️ Your Environment

Seen this way both in Firefox and Chrome.

```
OS:   Darwin 22.6.0 - darwin/arm64
node: v18.15.0
yarn: 1.22.19
cli:  0.22.9 (installed)
backstage:  1.16.0

Dependencies:
  @backstage/app-defaults                          1.4.1
  @backstage/backend-app-api                       0.4.5
  @backstage/backend-common                        0.19.1
  @backstage/backend-dev-utils                     0.1.1
  @backstage/backend-plugin-api                    0.5.4
  @backstage/backend-tasks                         0.5.4
  @backstage/catalog-client                        1.4.3
  @backstage/catalog-model                         1.4.1
  @backstage/cli-common                            0.1.12
  @backstage/cli-node                              0.1.2
  @backstage/cli                                   0.22.9
  @backstage/config-loader                         1.3.2
  @backstage/config                                1.0.8
  @backstage/core-app-api                          1.9.0
  @backstage/core-components                       0.12.5, 0.13.3
  @backstage/core-plugin-api                       1.5.3
  @backstage/dev-utils                             1.0.17
  @backstage/errors                                1.2.1
  @backstage/eslint-plugin                         0.1.3
  @backstage/integration-aws-node                  0.1.5
  @backstage/integration-react                     1.1.15
  @backstage/integration                           1.5.1
  @backstage/plugin-analytics-module-ga4           0.1.2
  @backstage/plugin-api-docs                       0.9.6
  @backstage/plugin-app-backend                    0.3.47
  @backstage/plugin-auth-backend                   0.18.5
  @backstage/plugin-auth-node                      0.2.16
  @backstage/plugin-catalog-backend                1.11.0
  @backstage/plugin-catalog-common                 1.0.15
  @backstage/plugin-catalog-graph                  0.2.32
  @backstage/plugin-catalog-graphql                0.3.22
  @backstage/plugin-catalog-import                 0.9.10
  @backstage/plugin-catalog-node                   1.4.0
  @backstage/plugin-catalog-react                  1.8.0
  @backstage/plugin-catalog                        1.12.0
  @backstage/plugin-events-node                    0.2.8
  @backstage/plugin-github-actions                 0.6.1
  @backstage/plugin-github-pull-requests-board     0.1.14
  @backstage/plugin-graphiql                       0.2.52
  @backstage/plugin-graphql-backend                0.1.37
  @backstage/plugin-home-react                     0.1.1
  @backstage/plugin-home                           0.5.4
  @backstage/plugin-org                            0.6.10
  @backstage/plugin-permission-common              0.7.7
  @backstage/plugin-permission-node                0.7.10
  @backstage/plugin-permission-react               0.4.14
  @backstage/plugin-proxy-backend                  0.2.41
  @backstage/plugin-scaffolder-backend             1.15.1
  @backstage/plugin-scaffolder-common              1.3.2
  @backstage/plugin-scaffolder-node                0.1.5
  @backstage/plugin-scaffolder-react               1.5.1
  @backstage/plugin-scaffolder                     1.14.1
  @backstage/plugin-search-backend-module-catalog  0.1.3
  @backstage/plugin-search-backend-module-pg       0.5.8
  @backstage/plugin-search-backend-module-techdocs 0.1.3
  @backstage/plugin-search-backend-node            1.2.3
  @backstage/plugin-search-backend                 1.3.3
  @backstage/plugin-search-common                  1.2.5
  @backstage/plugin-search-react                   1.6.3
  @backstage/plugin-search                         1.3.3
  @backstage/plugin-tech-radar                     0.6.6
  @backstage/plugin-techdocs-addons-test-utils     1.0.16
  @backstage/plugin-techdocs-backend               1.6.4
  @backstage/plugin-techdocs-module-addons-contrib 1.0.15
  @backstage/plugin-techdocs-node                  1.7.3
  @backstage/plugin-techdocs-react                 1.1.8
  @backstage/plugin-techdocs                       1.6.5
  @backstage/plugin-user-settings                  0.7.5
  @backstage/release-manifests                     0.0.9
  @backstage/test-utils                            1.4.1
  @backstage/theme                                 0.2.19, 0.4.1
  @backstage/types                                 1.1.0
  @backstage/version-bridge                        1.0.4
```

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by kunickiaj**
## Hey, I just made a Pull Request!
Fixes #19368 introduced in #17607 where borders of the cards in the customizable homepage arent rendered properly.
This restores the look of v1.14.

v1.14 and earlier
260580494-ca2d38cb-e861-4d00-ae3a-b

v1.15 with #17607 
260580489-c47bb332-aa48-4d33-89be-1

resolves #19368 

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by kunickiaj** [comment]
Please reopen and review.

**Review by awanlin — APPROVED** [review]
Looks good to me, thanks @kunickiaj 

@drodil any chance you can give this a quick look? Feel like you might have a good feel for the impact of this change

**Comment by drodil** [comment]
Don't think this breaks anything, the `overflow: hidden` was there to make sure that the widget content doesn't go over the container but I think the problem was in some cards instead the actual widget wrapper.

**Comment by awanlin** [comment]
Hi @Rugvip this is good to go just needs your approval and merge 🚀 

Thanks for following up @drodil 👍

**Review by Rugvip — APPROVED** [review]
@backstage/discoverability-maintainers PTAL 🙏

**Comment by awanlin** [comment]
Hi @aitherios, sorry to ping you directly but this is a small change and would be great to get it merged in but we need review from your team, please

**Comment by kunickiaj** [comment]
Following up here to see if we can get this merged please.

**Pull request merged**
